RoddyCore/src/de/dkfz/roddy/core/RuntimeService.groovy

```diff
@@ -160,15 +159,119 @@ public abstract class RuntimeService {
+        if(analysis.configuration.configurationValues.get("loadCohortDatasets", "false").toBoolean()) {
+            return loadCohortDatasetsWithFilter(analysis, pidFilters, suppressInfo)
+        } else {
+            loadStandardDatasetsWithFilter(analysis, pidFilters, suppressInfo)
+        }
+    }
+
+    List<DataSet> loadStandardDatasetsWithFilter(Analysis analysis, List<String> pidFilters, boolean suppressInfo = false) {
         if (pidFilters == null || pidFilters.size() == 0 || pidFilters.size() == 1 && pidFilters.get(0).equals("[ALL]")) {
             pidFilters = Arrays.asList("*");
         }
         List<DataSet> listOfDataSets = getListOfPossibleDataSets(analysis);
         return selectDatasetsFromPattern(analysis, pidFilters, listOfDataSets, suppressInfo);
     }
 
-    public List<DataSet> selectDatasetsFromPattern(Analysis analysis, List<String> pidFilters, List<DataSet> listOfDataSets, boolean suppressInfo) {
+    /**
+     * In difference to the normal behaviour, the method / workflow loads with supercohorts and cohorts thus using a different
+     * dataset id format on the command line:
+     * run p@a s:[c:DS0;DS1;...;DSn|c:DS2;...]
+     *
+     * comma separation stays
+     * a cohort must be marked with c: in front of it. That is just to make the call clear and has no actual use!
+     * entries in a cohort are separated by semicolon!
+     * Remark, that order matters! The first entry of a cohor is the major cohort dataset which will e.g. be used for file
+     * output, if applicable.
+     *
+     * TODO Identify cases, where the matching mechanism should fail! Like e.g. when a dataset is missing but requested.
+     * - Using wildcards? At least one dataset should be matched for each wildcard entry.
+     * - Using no wildcards? There must be one dataset match.
+     *
+     * @param analysis
+     * @param pidFilters
+     * @param suppressInfo
+     * @return
+     */
+    List<DataSet> loadCohortDatasetsWithFilter(Analysis analysis, List<String> pidFilters, boolean suppressInfo) {
+        List<DataSet> listOfDataSets = getListOfPossibleDataSets(analysis);
+        List<SuperCohortDataSet> datasets
+
+        boolean foundFaulty = false;
+        pidFilters.each {
+            boolean faulty = !validateDataSetLoadingString(it)
+            if (faulty)
+                logger.severe("The pid string ${it} is malformed.")
+            foundFaulty |= faulty
+        }
+        if (foundFaulty) {
+            logger.severe("The dataset list you provided contains errors, Roddy will not start jobs.")
+            return [];
+        }
+
+        boolean error = false
+
+        // Checks are all done, now get the datasets..
+        datasets = pidFilters.collect {
+            String superCohortDescription ->
+                // Remove leading s[ and trailing ], split by |
+                List<CohortDataSet> cohortDatasets = superCohortDescription[2..-2].split("[|]").collect {
```

Comment:
The method checks the string before. If it is invalid, this part of the code is not executed.